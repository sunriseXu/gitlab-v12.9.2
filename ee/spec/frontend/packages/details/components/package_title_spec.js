import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PackageTitle from 'ee/packages/details/components/package_title.vue';
import PackageTags from 'ee/packages/shared/components/package_tags.vue';
import {
  conanPackage,
  mavenFiles,
  mavenPackage,
  mockTags,
  npmFiles,
  npmPackage,
  nugetPackage,
} from '../../mock_data';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('PackageTitle', () => {
  let wrapper;
  let store;

  function createComponent(packageEntity = mavenPackage, packageFiles = mavenFiles) {
    store = new Vuex.Store({
      state: {
        packageEntity,
        packageFiles,
      },
      getters: {
        packageTypeDisplay: ({ packageEntity: { package_type: type } }) => type,
      },
    });

    wrapper = shallowMount(PackageTitle, {
      localVue,
      store,
    });
  }

  const packageType = () => wrapper.find({ ref: 'package-type' });
  const packageSize = () => wrapper.find({ ref: 'package-size' });
  const packageTags = () => wrapper.find(PackageTags);

  afterEach(() => {
    wrapper.destroy();
  });

  describe('renders', () => {
    it('without tags', () => {
      createComponent();

      expect(wrapper.element).toMatchSnapshot();
    });

    it('with tags', () => {
      createComponent({ ...mavenPackage, tags: mockTags });

      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe.each`
    packageEntity   | expectedResult
    ${conanPackage} | ${'conan'}
    ${mavenPackage} | ${'maven'}
    ${npmPackage}   | ${'npm'}
    ${nugetPackage} | ${'nuget'}
  `(`package type`, ({ packageEntity, expectedResult }) => {
    beforeEach(() => createComponent(packageEntity));

    it(`${packageEntity.package_type} should render from Vuex getters ${expectedResult}`, () => {
      expect(packageType().text()).toBe(expectedResult);
    });
  });

  describe('calculates the package size', () => {
    it('correctly calulates when there is only 1 file', () => {
      createComponent(npmPackage, npmFiles);

      expect(packageSize().text()).toBe('200 bytes');
    });

    it('correctly calulates when there are multiple files', () => {
      createComponent();

      expect(packageSize().text()).toBe('300 bytes');
    });
  });

  describe('package tags', () => {
    it('displays the package-tags component when the package has tags', () => {
      createComponent({
        ...npmPackage,
        tags: mockTags,
      });

      expect(packageTags().exists()).toBe(true);
    });

    it('does not display the package-tags component when there are no tags', () => {
      createComponent();

      expect(packageTags().exists()).toBe(false);
    });
  });
});
