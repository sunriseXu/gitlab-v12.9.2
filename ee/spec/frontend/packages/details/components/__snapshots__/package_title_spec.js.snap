// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageTitle renders with tags 1`] = `
<div
  class="flex-column"
>
  <h1
    class="gl-font-size-20 prepend-top-8 append-bottom-4"
  >
    Test package
  </h1>
   
  <div
    class="d-flex align-items-center text-secondary"
  >
    <gl-icon-stub
      class="append-right-8"
      name="eye"
      size="16"
    />
     
    <gl-sprintf-stub
      message="v%{version} published %{timeAgo}"
    />
  </div>
   
  <div
    class="d-flex flex-wrap align-items-center append-bottom-8"
  >
    <div
      class="d-flex align-items-center append-right-default"
    >
      <gl-icon-stub
        class="text-secondary append-right-8"
        name="package"
        size="16"
      />
       
      <span
        class="font-weight-bold"
      >
        maven
      </span>
    </div>
     
    <div
      class="d-flex align-items-center append-right-default"
    >
      <package-tags-stub
        tagdisplaylimit="1"
        tags="[object Object],[object Object],[object Object],[object Object]"
      />
    </div>
     
    <div
      class="d-flex align-items-center append-right-default"
    >
      <gl-icon-stub
        class="text-secondary append-right-8"
        name="disk"
        size="16"
      />
       
      <span
        class="font-weight-bold"
      >
        300 bytes
      </span>
    </div>
  </div>
</div>
`;

exports[`PackageTitle renders without tags 1`] = `
<div
  class="flex-column"
>
  <h1
    class="gl-font-size-20 prepend-top-8 append-bottom-4"
  >
    Test package
  </h1>
   
  <div
    class="d-flex align-items-center text-secondary"
  >
    <gl-icon-stub
      class="append-right-8"
      name="eye"
      size="16"
    />
     
    <gl-sprintf-stub
      message="v%{version} published %{timeAgo}"
    />
  </div>
   
  <div
    class="d-flex flex-wrap align-items-center append-bottom-8"
  >
    <div
      class="d-flex align-items-center append-right-default"
    >
      <gl-icon-stub
        class="text-secondary append-right-8"
        name="package"
        size="16"
      />
       
      <span
        class="font-weight-bold"
      >
        maven
      </span>
    </div>
     
    <!---->
     
    <div
      class="d-flex align-items-center append-right-default"
    >
      <gl-icon-stub
        class="text-secondary append-right-8"
        name="disk"
        size="16"
      />
       
      <span
        class="font-weight-bold"
      >
        300 bytes
      </span>
    </div>
  </div>
</div>
`;
